src/components/FormInput/FormInput.tsx

```diff
@@ -0,0 +1,58 @@
+import React from 'react';
+import TextField from '@mui/material/TextField';
+import Grid from '@mui/material/Grid';
+import { IGrid } from 'types';
+import { IInput } from '../Form/Form.type';
+
+interface FormInputProps {
+  value: any;
+  onChange: (id: string, value: any) => void;
+  config: IInput['config'];
+  error?: string;
+  grid: IGrid;
+}
+
+const FormInput: React.FC<FormInputProps> = ({
+  value,
+  onChange,
+  config,
+  error,
+  grid,
+}) => {
+  let component = null;
+
+  switch (config.component) {
+    case 'text-field':
```

Comment:
Probably this is a simple case scenario, but when we have a large application I recommend to create a custom data type to evaluate `case` sentences.